main/src/main/scala/sbt/Defaults.scala

```diff
@@ -716,10 +716,15 @@ object Defaults extends BuildCommon {
 
   def doClean(clean: Seq[File], preserve: Seq[File]): Unit =
     IO.withTemporaryDirectory { temp =>
+      def cleanLeaveRoot(root: File) = {
+        val (dirs, files) = (root.*** --- root).get.partition(_.isDirectory)
+        IO.delete(files)
+        IO.delete(dirs)
```

Comment:
For an unknown reason, I thought that trying to delete an already deleted file could end up badly, but it won't.